js/src/modals/SMSVerification/store.js

```diff
@@ -0,0 +1,236 @@
+export const GATHERED_DATA = 'gathered-data';
+export const POSTING_REQUEST = 'posting-request';
+export const POSTED_REQUEST = 'posted-request';
+export const REQUESTING_SMS = 'requesting-sms';
+export const REQUESTED_SMS = 'requested-sms';
+export const QUERY_CODE = 'query-code';
+export const POSTING_CONFIRMATION = 'posting-confirmation';
+export const POSTED_CONFIRMATION = 'posted-confirmation';
+export const DONE = 'done';
+
+export default class VerificationStore {
+  @observable step = null;
+  @observable error = null;
+
+  @observable fee = null;
+  @observable isVerified = null;
+  @observable hasRequested = null;
+  @observable consentGiven = false;
+  @observable number = '';
+  @observable requestTx = null;
+  @observable code = '';
+  @observable confirmationTx = null;
+
+  @computed get isCodeValid () {
+    return validCode.test(this.code);
+  }
+  @computed get isNumberValid () {
+    return phone.isValidNumber(this.number);
+  }
+
+  @computed get isStepValid () {
+    if (this.step === DONE) {
+      return true;
+    }
+    if (this.error) {
+      return false;
+    }
+
+    if (this.step === GATHERED_DATA) {
+      return this.fee && this.isVerified === false && this.isNumberValid && this.consentGiven;
+    }
+    if (this.step === REQUESTED_SMS) {
+      return this.requestTx;
+    }
+    if (this.step === QUERY_CODE) {
+      return this.isCodeValid;
+    }
+    if (this.step === POSTED_CONFIRMATION) {
+      return this.confirmationTx;
+    }
+    return false;
+  }
+
+  @action setNumber = (number) => {
+    this.number = number;
+  }
+  @action setConsentGiven = (consentGiven) => {
+    this.consentGiven = consentGiven;
+  }
+  @action setCode = (code) => {
+    this.code = code;
+  }
+
+  constructor (api, account) {
+    this.api = api;
+    this.account = account;
+    this.contract = api.newContract(ABI, contract);
+
+    autorun(() => {
+      if (this.error) {
+        console.error('sms verification: ' + this.error);
+      }
+    });
+  }
+
+  @action gatherData = () => {
+    const { contract, account } = this;
+    this.step = GATHERING_DATA;
+
+    const fee = contract.instance.fee.call()
+      .then((fee) => {
+        this.fee = fee;
+      })
+      .catch((err) => {
+        this.error = 'Failed to fetch the fee: ' + err.message;
+      });
+
+    const isVerified = checkIfVerified(contract, account)
+      .then((isVerified) => {
+        this.isVerified = isVerified;
+      })
+      .catch((err) => {
+        this.error = 'Failed to check if verified: ' + err.message;
+      });
+
+    const hasRequested = checkIfRequested(contract, account)
+      .then((txHash) => {
+        this.hasRequested = !!txHash;
+        if (txHash) {
+          this.requestTx = txHash;
+        }
+      })
+      .catch((err) => {
+        this.error = 'Failed to check if requested: ' + err.message;
+      });
+
+    Promise.all([ fee, isVerified, hasRequested ])
+    .then(() => {
```

Comment:
indent. Consider .all on same level. (As it stands there is stylign inconsistency with the Promise chain just above this one.)